e2e_testing/torchscript/histogram_binning_calibration.py

```diff
@@ -0,0 +1,103 @@
+
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import torch
+
+from torch_mlir_e2e_test.torchscript.framework import TestUtils
+from torch_mlir_e2e_test.torchscript.registry import register_test_case
+from torch_mlir_e2e_test.torchscript.annotations import annotate_args, export
+
+
+# ==============================================================================
+
+class HistogramBinningCalibrationByFeature(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        # HistogramBinningCalibrationByFeature
+        self._num_segments = 42
+        self._num_bins = 5000
+        _num_interval = (self._num_segments + 1) * self._num_bins
+        _lower_bound = 0
+        _upper_bound = 1
+        l, u = _lower_bound, _upper_bound
+        w = (u - l) / self._num_bins
+        self.step = w
+        self.register_buffer("_boundaries", torch.arange(l + w, u - w / 2, w))
+        self.register_buffer(
+            "_bin_num_examples",
+            torch.empty([_num_interval], dtype=torch.float64).fill_(0.0),
+        )  # ConstantFill
+        self.register_buffer(
+            "_bin_num_positives",
+            torch.empty([_num_interval], dtype=torch.float64).fill_(0.0),
+        )  # ConstantFill
+        self.register_buffer("_bin_ids", torch.arange(_num_interval))
+        self.positive_weight = torch.tensor([0.4])
+        self.bin_ctr_in_use_after = 0
+        self.bin_ctr_weight_value = 0.9995
+        self.oneminusbin_ctr_weight_value = 0.0005
+        self._iteration = 0
+
+    @export
+    @annotate_args([
+        None,
+        ([5000], torch.int32, True),
+        ([5001], torch.int32, True),
+        ([5000], torch.float32, True),
+    ])
+    def forward(self, segment_value, segment_lengths, logit):
+        origin_prediction = torch.sigmoid(
+            logit + torch.log(self.positive_weight))
+        dense_segment_value = torch.zeros(logit.numel(), dtype=torch.int32)
+        validoffsets = torch.gt(
+            segment_lengths[1:self._num_bins+1], segment_lengths[0:self._num_bins])
+        gathered_segment_values = (
+            segment_value[segment_lengths[0:self._num_bins].long()]+1).int()
+        dense_segment_value = torch.where(
+            validoffsets, gathered_segment_values, dense_segment_value)
+        zeros = torch.empty_like(
+            dense_segment_value, dtype=torch.int32).fill_(0)  # ConstantFill
+        isnotvalid = torch.gt(dense_segment_value, self._num_segments)  # GT

```

Comment:
is this autogenerated? can you mention how it is generated?

Reply:
I used autopep8 
` autopep8 -i e2e_testing/torchscript/histogram_binning_calibration.py`

Reply:
Should I mention how I generated the  formatting as a comment in the file or is this resolved?

Reply:
Ahh, this is behind some layers that I dont have access to but I manually edited a lot of the original script anyway. @powderluv do you want to ellaborate on this?

Reply:
That is nothing special and is just comments. Here is the source of the source file from @harsh-nod :

https://github.com/pytorch/FBGEMM/pull/804/files

Reply:
Can we remove those autogenerated comments at least? I don't think they add much value.

Reply:
Yes, I can do that